discovery-provider/src/challenges/challenge_event.py

```diff
@@ -11,3 +11,5 @@ class ChallengeEvent(str, enum.Enum):
     favorite = "favorite"
     track_listen = "track_listen"
     track_upload = "track_upload"
+    referral_signup = "referral_signup"  # Fired for the referring user
```

Comment:
If I could hook into the updater to modify the event before the user challenges get created, I might be able to get away with a single event here and two different listeners. But not sure that's better...

Reply:
I see what you're saying - would be nice to have some flexibility for how the user_challenges are fetched (have the referred user challenge by fetched by the referred extra data). But I don't mind two events too much, I think not worth undertaking this change